tensorflow_addons/metrics/f1_test.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests F1 metrics."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow_addons.metrics import F1Score
+from tensorflow_addons.utils import test_utils
+
+
+@test_utils.run_all_in_graph_and_eager_modes
+class F1ScoreTest(tf.test.TestCase):
+    def test_config(self):
+        f1_obj = F1Score(name='f1_score', num_classes=3)
+        self.assertEqual(f1_obj.name, 'f1_score')
+        self.assertEqual(f1_obj.dtype, tf.float32)
+        self.assertEqual(f1_obj.num_classes, 3)
+        # Check save and restore config
+        f1_obj2 = F1Score.from_config(f1_obj.get_config())
+        self.assertEqual(f1_obj2.name, 'f1_score')
+        self.assertEqual(f1_obj2.dtype, tf.float32)
+        self.assertEqual(f1_obj2.num_classes, 3)
+
+    def initialize_vars(self):
+        f1_obj = F1Score(num_classes=3, average='micro')
+        f1_obj1 = F1Score(num_classes=3, average='macro')
+        f1_obj2 = F1Score(num_classes=3, average='weighted')
+
+        self.evaluate(tf.compat.v1.variables_initializer(f1_obj.variables))
+        self.evaluate(tf.compat.v1.variables_initializer(f1_obj1.variables))
+        self.evaluate(tf.compat.v1.variables_initializer(f1_obj2.variables))
+        return f1_obj, f1_obj1, f1_obj2
+
+    def initialize_vars_none(self):
```

Comment:
Combine `f1_none` into `initialize_vars`. It's better than having a separate method that does the exact same thing.